### DISCUSSION TITLE
Fetch all is very slow

### DISCUSSION THREAD

**Discussion by rasqual** [dbeaver/dbeaver#22718] [Q&A]
New to DBeaver, hoping this is me unaware of a setting, vs. a bug.

We've been using Aqua Data Studio. ADS fetches all rows lightning fast, DBeaver not so much. The query I have in front of me takes 27 milliseconds in ADS, but 21 seconds for DBeaver. 

I'm not used to seeing progress bars where I've been spoiled by getting instant results. 

Heck, fetching the first 200 rows is taking multiple times longer than getting all rows in ADS. 

What am I likely getting wrong?

MS SQL, Community (for now) DBeaver.

**Comment by LonwoLonwo** [comment]
Hello @rasqual 

Fetching can depend on the driver version.
Could you please check your driver version? You can find it in your Connection Settings -> Test Connection (or Driver settings)
If you know your driver version in ADS, you can provide it too.

**Comment by rasqual** [comment]
In DBeaver
Dialect: Microsoft SQL Server  DBMS: Microsoft SQL Server (ver. 15.00.2104) Case sensitivity: plain=mixed, delimited=mixed Driver: jTDS Type 4 JDBC Driver for MS SQL Server and Sybase (ver. 1.3.1, JDBC3.0) 

In ADS
Server Product Name:  Microsoft SQL Server
Server Product Version:  15.00.2104
Driver Name:  jTDS Type 4 JDBC Driver for MS SQL Server and Sybase
Driver Version: 1.2.5
Driver Class: net.sourceforge.jtds.jdbc.Driver
Query Server Information: Microsoft SQL Server 2019 (RTM-GDR) (KB5029377) - 15.0.2104.1 (X64) 
	Aug 16 2023 00:09:21 
	Copyright (C) 2019 Microsoft Corporation
	Standard Edition (64-bit) on Windows Server 2019 Standard 10.0  (Build 17763: ) (Hypervisor)

**Reply by LonwoLonwo** [reply]
> Driver Version: 1.2.5

You can decrease your driver version in your connection settings -> driver settings -> libraries
